spec/rubocop/cop/rspec/overwriting_setup_spec.rb

```diff
@@ -0,0 +1,46 @@
+RSpec.describe RuboCop::Cop::RSpec::OverwritingSetup do
+  subject(:cop) { described_class.new }
+
+  it 'finds overwriten `let`' do
+    expect_violation(<<-RUBY)
+      RSpec.describe User do
+        let(:a) { a }
+        let(:a) { b }
+        ^^^^^^^^^^^^^ `a` is already defined.
+      end
+    RUBY
+  end
+
+  it 'finds overwriten `subject`' do
+    expect_violation(<<-RUBY)
+      RSpec.describe User do
+        subject(:a) { a }
+
+        let(:a) { b }
+        ^^^^^^^^^^^^^ `a` is already defined.
```

Comment:
This is pretty cool! Neat that it checks for overriding `subject`s and `let`s and vice versa rather than just one of each kind.